packages/core/lib/debug/printer.js

```diff
@@ -209,11 +210,43 @@ class DebugPrinter {
       this.config.logger.log(DebugUtils.formatStack(step.stack));
       this.config.logger.log("");
     }
-    this.config.logger.log(
-      DebugUtils.formatInstruction(traceIndex + 1, totalSteps, instruction)
-    );
-    this.config.logger.log(DebugUtils.formatPC(step.pc));
+
+    this.config.logger.log("Instructions:");
+    this.config.logger.log("");
+
+    // printout instructions
+    // add an ellipse if there exist additional instructions before
+    const previousInstructions = 3;
+    const upcomingInstructions = 3;
+    const currentIndex = instruction.index;
+
+    if (currentIndex - previousInstructions >= 0) {
+      this.config.logger.log("...");
+    }
+    // printout 3 previous instructions
+    for (let i = Math.max(currentIndex - previousInstructions, 0);
+      i < currentIndex;
+      i++) {
+      this.config.logger.log(DebugUtils.formatInstruction(instructions[i]));
+    }
+
+    // printout current instruction
+    this.config.logger.log(DebugUtils.formatCurrentInstruction(instruction));
+
+    // printout 3 upcoming instructions
+    for (let i = Math.min(currentIndex + 1, instructions.length);
+      i <= currentIndex + upcomingInstructions;
+      i++) {
```

Comment:
This is definitely not the style `prettier` would use, but, to hell with it, I'm not going to hold things up over that!  It shouldn't be your job to deal with that anyway; we need to get Husky fixed and then you won't ever have to worry about this again! (@cds-amal)

(This applies to the other `for` loop as well, obviously.)

Reply:
Ok, let's leave this for now and wait for Husky to work and fix all.